### ISSUE TITLE
Is there any introduction of the gpu-next thing?

### ISSUE THREAD

**Issue by laichiaheng** [mpv-player/mpv#9506]
I saw lots of issues about gpu-next here, what is it actually?

Can a non-developer like me understand what it does?

Will this support VAAPI in the future?

By the way, I don't see this --vo option in `mpv-full-build-git` from AUR

**Comment by LaserEyess** [comment]
1.  gpu-next is an experimental vo, you can see a lot of the discussion behind it here: https://github.com/mpv-player/mpv/pull/8799. 
2. I don't know what you want to understand about it as a "non-developer", but basically the idea is you do `--vo=gpu-next` and then it Just Werks:tm: (in theory, but clearly given the issues it does not Just Werk quite yet). You might encounter bugs and performance issues if you use it, so really it's only for testing right now and you shouldn't rely on it to have 1:1 compatibility with ``--vo=gpu``. If you want to use it and find issues, you can report them. 
3. hwdec support is planned but not implemented yet. Again, this is experimental.

Also, build issues should be brought up with that PKGBUILD maintainer, though keep in mind that right now gpu-next requires the latest libplacebo, so if you don't have that then it will disable gpu-next at compile time.

**Comment by hooke007** [comment]
> Can a non-developer like me understand what it does?

Like the another external video renderer. Not exact but easy to understand.

**Comment by haasn** [comment]
Heads-up: I wrote [a brief wiki article](https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU) explaining the "important to know" information.

**Comment by simonmysun** [comment]
Hi, the option seems also not in https://archlinux.org/packages/community/x86_64/mpv/ on my device. The current version is 1:0.34.1-3. 

other environmental libraries I have: 

* mesa-21.3.6-1
* nvidia-utils-510.54-1
* vulkan-intel-21.3.6-1
* libplacebo-4.192.1-1
* ffmpeg-2:5.0-4

when I run `mpv --vo=help` I get:
```
Available video outputs:
  libmpv           render API for libmpv
  gpu              Shader-based GPU Renderer
  vdpau            VDPAU with X11
  wlshm            Wayland SHM video output (software scaling)
  xv               X11/Xv
  vaapi            VA API with X11
  x11              X11 (software scaling)
  null             Null video output
  image            Write video frames to image files
  tct              true-color terminals
  drm              Direct Rendering Manager (software scaling)
```
and if I use --vo=gpu-next it shouts:
```
[vo] Video output gpu-next not found!
Error opening/initializing the selected video_out (--vo) device.
```

What's the condition to get gpu-next available?

**Comment by LaserEyess** [comment]
[File a bug with Arch](https://bugs.archlinux.org/newtask?project=5&product_category=33&item_summary=%5Bmpv%5D+PLEASE+ENTER+SUMMARY)

**Comment by simonmysun** [comment]
Thank you. I tried building from latest git master but still no gpu-next. 

Some friend told me on 23 Feb. 2022 (I haven't validate): 

> The libplacebo in official repo seems missing libplacebo-v4 support. You will need libplacebo-git in AUR and replace ffmpeg with ffmpeg-vulkan

I decide to wait for the official support and anyone still having interest may give a try. If it doesn't work please tell me and I will remove this.

**Comment by qmega** [comment]
You need libplacebo-git, but the normal ffmpeg in the Arch official repos (extra) works, provided you've updated in the last week or so (it's currently at version `2:5.0-4`).

(Right now gpu-next depends on unreleased libplacebo API, so you need libplacebo-git on basically any distro.)

**Comment by simonmysun** [comment]
I see, thx!

**Comment by Bananaman** [comment]
After researching:

- https://github.com/mpv-player/mpv/wiki/GPU-Next-vs-GPU
- Various tickets.

Conclusion:

- `gpu-next` requires the latest (unreleased) git version of `libplacebo` and the latest (unreleased) git version of `mpv`.
- It's still in testing/fixing/development stages and isn't recommended unless people want to test unfinished, bleeding-edge code.
- There's no `libmpv` support for it yet, which means that players that embed mpv's core in a different GUI will not be able to use GPU-Next yet. It only works in the official mpv app.
- The main benefits of GPU-Next are that it has much more efficient rendering code, and that it supports HDR tonemapping, including HDR/DV, and including HDR/DV to SDR (useful for Linux users since Linux lacks HDR support).
- People who switch to GPU-Next will need to edit a lot of things in their `mpv.conf` file, since plenty of older filters and features have been removed in GPU-Next. The wiki page above lists a lot about it.
- In short: It's a great new development but it's not finished yet, so regular users shouldn't feel any "pressure" to update to it. Wait for the general, public release when libplacebo and mpv are ready for public use. You'll have less hassle and less need for mpv.conf editing and workarounds if you wait. Think about this too: Everyone has been living with and loving regular `vo=gpu` for many, many years now, so there's no rush to "upgrade" to the next still-in-development thing! Just chill, unless you're here to test the broken features and report bugs (which is a noble thing of course!).

**Comment by Dudemanguy** [comment]
>Haha yeah the wiki was edited 1 hour after my post

By me. ;)

**Comment by Bananaman** [comment]
@Dudemanguy Ohhhhh! Awesome! :rofl:

**Comment by hooke007** [comment]
> so the render api will work with it 

I'm not sure. Is that means gpu-next would work with `vo=libmpv` ?

**Comment by neuromagus** [comment]
and who can tell me what is preferable in Wlroots WM? ```gpu-next``` or ```dmabuf-wayland```? And what is generally better in terms of rendering and laptop battery? If I'm not mistaken, Wayland does not use opengl, but uses Vulkan?

**Comment by LaserEyess** [comment]
Please use [discussions](https://github.com/mpv-player/mpv/discussions) instead, this is a completely unreleated issue and your questions have nothing to do with it.

**Comment by neuromagus** [comment]
thanks a lot
